src/server/rdb_load.cc

```diff
@@ -198,6 +199,66 @@ int ziplistPairsConvertAndValidateIntegrity(const uint8_t* zl, size_t size, unsi
   return ret;
 }
 
+/* callback for to check the listpack doesn't have duplicate records */
+static int _lpEntryValidation(unsigned char* p, unsigned int head_count, void* userdata) {
+  struct cb_args {
+    int pairs;
+    long count;
+    dict* fields;
+  }* data = (cb_args*)userdata;
+
+  if (data->fields == NULL) {
+    data->fields = dictCreate(&hashDictType);
+    dictExpand(data->fields, data->pairs ? head_count / 2 : head_count);
+  }
+
+  /* If we're checking pairs, then even records are field names. Otherwise
+   * we're checking all elements. Add to dict and check that's not a dup */
+  if (!data->pairs || ((data->count) & 1) == 0) {
+    unsigned char* str;
+    int64_t slen;
+    unsigned char buf[LP_INTBUF_SIZE];
+
+    str = lpGet(p, &slen, buf);
+    sds field = sdsnewlen(str, slen);
+    if (dictAdd(data->fields, field, NULL) != DICT_OK) {
+      /* Duplicate, return an error */
+      sdsfree(field);
+      return 0;
+    }
+  }
+
+  (data->count)++;
+  return 1;
+}
+
+/* Validate the integrity of the listpack structure.
+ * when `deep` is 0, only the integrity of the header is validated.
+ * when `deep` is 1, we scan all the entries one by one.
+ * when `pairs` is 0, all elements need to be unique (it's a set)
+ * when `pairs` is 1, odd elements need to be unique (it's a key-value map) */
+int lpValidateIntegrityAndDups(unsigned char* lp, size_t size, int deep, int pairs) {
```

Comment:
hmm, we never call `lpValidateIntegrityAndDups` with deep =1 , so lets eliminate the code we never run.
since we use it only for loading from redis, I am fine with reducing the amount of checks.